cgen error with calling a function accepting `...Type`

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

<details>
<summary>V version: V 0.4.12 8a1efb3, press to see full `v doctor` output</summary>

|V full version      |V 0.4.12 8a1efb3fe0a586b84e2acc6dd761cfb776aeca3a
|:-------------------|:-------------------
|OS                  |linux, Debian GNU/Linux 12 (bookworm) (WSL 2)
|Processor           |4 cpus, 64bit, little endian, Intel(R) Core(TM) i5-7200U CPU @ 2.50GHz
|Memory              |6.04GB/7.69GB
|                    |
|V executable        |/home/zev/v/v
|V last modified time|2025-11-21 18:19:31
|                    |
|V home dir          |OK, value: /home/zev/v
|VMODULES            |OK, value: /home/zev/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /home/zev/vlang/veblog/vandoc
|                    |
|Git version         |git version 2.39.5
|V git status        |8a1efb3f
|.git/config present |true
|                    |
|cc version          |cc (Debian 12.2.0-14+deb12u1) 12.2.0
|gcc version         |gcc (Debian 12.2.0-14+deb12u1) 12.2.0
|clang version       |N/A
|tcc version         |tcc version 0.9.28rc 2025-02-13 HEAD@f8bd136d (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 696c1d84
|emcc version        |N/A
|glibc version       |ldd (Debian GLIBC 2.36-9+deb12u10) 2.36

</details>

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg  vandoc.v && vandoc`
```v
module vandoc
import os
import v.util.diff

pub fn get_pandoc() !string {
	return os.find_abs_path_of_executable("pandoc") or {
		os.find_abs_path_of_executable("pandoc.exe") or {
			error("Pandoc not installed or not in path.")
		}
	}
}

pub fn convert(from string, to string, file string, flags ...ArgFlag) ! {
	mut command := "${get_pandoc() or { eprintln(err.msg()); return } } ${file} -f ${from} -t ${to} "
	for flag in flags {
		command += "--${flag.flag.name} ${if flag.has_arg {flag.arg+' ' } else {''} }"
	}
	os.execute_or_exit(command)
}

pub fn live_convert(from string, to string, file string, flags ...ArgFlag) ! {
	for true {
		os.cp(file, "tmp/${file}") or {eprintln(err.msg()); return}
		convert("tmp/${file}",to,file,flags) or {eprintln(err.msg()); return}
		for diff.compare_files("tmp/${file}",file)! != "" {}
	}
}

```

**What did you see?**
```
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/veblog.01KAX3VQZ3Q15JA7EVXW1V7THM.tmp.c:8719: error: cannot convert 'struct array' to 'unsigned char *'
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error:
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

**What did you expect to see?**

No compiler error, the issue seems to be related to `...Type` handling because when it is modified to be `[]ArgFlag` the issue dissapears
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -2612,7 +2612,7 @@ fn (mut g Gen) call_args(node ast.CallExpr) {
 					}
 				} else {
 					// passing variadic arg to another call which expects same array type
-					if args.len == 1
+					if variadic_count == 1
 						&& ((args[arg_nr].typ.has_flag(.variadic) && args[arg_nr].typ == varg_type)
 						|| (varg_type.has_flag(.variadic)
 						&& args[arg_nr].typ == varg_type.clear_flag(.variadic))) {

--- a/vlib/v/tests/fns/variadic_fn_chain_test.v
+++ b/vlib/v/tests/fns/variadic_fn_chain_test.v
@@ -1,0 +1,9 @@
+fn called(s string, l ...string) {}
+
+fn caller(s string, l ...string) {
+	called(s, l)
+}
+
+fn test_main() {
+	caller('')
+}

